File: 75 days Challenge Leetcode/Array+String/Day-01/Solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        # Get the lengths of both input strings
        len1 = len(word1)
        len2 = len(word2)
        
        # Initialize an empty list to store the merged result (lists are more efficient for multiple concatenations)
        result = []
        
        # Find the smaller length between word1 and word2
        minlength = min(len1, len2)

        # Loop through both strings alternately until the end of the shorter one
        for i in range(minlength):
            # Append characters alternately from both strings
            result.append(word1[i])
            result.append(word2[i])

        # Append remaining characters from the longer string
        if len1 > len2:
            result.append(word1[minlength:])  # Remaining characters from word1
        else:
            result.append(word2[minlength:])  # Remaining characters from word2

        # Join the list into a single string and return
        return ''.join(result)

File: 75 days Challenge Leetcode/Array+String/Day-01/Solution.cpp
class Solution {
public:

    string mergeAlternately(string word1, string word2) {

        int len1 = word1.length();
        int len2 = word2.length();
        
    // to store the final result
        string result = "";

        // Determine the smaller length between the two strings
        int minlength = len1 < len2 ? len1 : len2;

        // Loop through the characters of both strings up to the length of the shorter one
        for (int i = 0; i < minlength; i++) {
            // Append the character from word1 followed by the character from word2
            result += word1[i];
            result += word2[i];
        }

        // Append the remaining characters from the longer string (if any)
        // If word1 is longer, append the rest of word1
        // If word2 is longer, append the rest of word2
        return (len1 > len2) ? result += word1.substr(minlength)
                            : result += word2.substr(minlength);
    }
};
